impl/src/main/java/dev/paseto/jpaseto/impl/crypto/JcaV1PublicCryptoProvider.java

```diff
@@ -67,12 +70,27 @@ public boolean verify(byte[] message, byte[] footer, byte[] signature, PublicKey
     }
 
     private Signature pssSignature() {
-        try {
-            Signature rsaSignature = Signature.getInstance("RSASSA-PSS");
-            rsaSignature.setParameter(new PSSParameterSpec("SHA-384", "MGF1", new MGF1ParameterSpec("SHA-384"), 48, 1));
-            return rsaSignature;
-        } catch (NoSuchAlgorithmException | InvalidAlgorithmParameterException e) {
-            throw new PasetoSignatureException("Could not load signature algorithm 'RSASSA-PSS' ensure you are using jpaseto-bouncy-castle.jar or Java 11+", e);
+        Signature rsaSignature;
+        if (IS_IN_BC_FIPS_MODE) {
```

Comment:
In my opinion the code flow and error handling is clearer if the BC-FIPS and non-BC-FIPS code paths are kept separate, although there is a little bit of code duplication

Reply:
All looks great.  Thanks Michael!  (And I wonder if there could be flexibility built into the JPaseto side -- perhaps as a build flag or via properties) to designate some of those magic values in case the upcoming bc-fips 2.0.0 would change that static string again.)

Reply:
What is the key in v2?
We _should_ be able to look up the keys in the BC provider and pick one or the other

Reply:
BC-FIPS 2.0.0 hasn't been released yet as far as I know -- I think this is primarily a future-proofing concern in case the signature algorithm name changes again (which I hope was just a quirk of the different requirements for BC-FIPS vs normal BC instead of something that will happen regularly).

It seems to me that it would be valuable to keep this point in the back of our minds for now and then once 2.0.0 is released we can revisit and see if we need to add that extra complexity/intelligence or not.  It seems unlikely that the name would change again but it's definitely a good point to consider if you guys want to extend support to other BC-FIPS releases.

I think the other "magic" values Ash is calling out are the parameters to the `PSSParameterSpec` like the salt length and trailer field constants.  It seems those parameters are not explicitly defined in the PASETO RFC so I wasn't sure if there were reasons you picked the values you did, but making those configurable (if allowed by the PASETO spec) might be interesting.